py/server/deephaven/experimental/iceberg.py

```diff
@@ -0,0 +1,300 @@
+
+    def namespaces(self, namespace: Optional[str] = None) -> Sequence[str]:
+        """
+        Returns the list of namespaces in the catalog.
+
+        Args:
+           namespace (Optional[str]): the higher-level namespace from which to list namespaces; if omitted, the
+                top-level namespaces are listed.
+        Returns:
+            Sequence[str]: the list of namespaces.
+        """
+        if namespace is not None:
+            return _j_object_list_to_str_list(self.j_object.listNamespaces(namespace))
+        return _j_object_list_to_str_list(self.j_object.listNamespaces())
+
+    def namespaces_as_table(self, namespace: Optional[str] = None) -> Table:
+        """
+        Returns the namespaces in the catalog as a Deephaven table.
+
+        Args:
+            namespace (Optional[str]): the higher-level namespace from which to list namespaces; if omitted, the
+                top-level namespaces are listed.
+
+        Returns:
+            a table containing the namespaces.
+        """
+
+        if namespace is not None:
+            return Table(self.j_object.listNamespaces(namespace))
+        return Table(self.j_object.listNamespacesAsTable())
+
+    def tables(self, namespace: str) -> Sequence[str]:
+        """
+        Returns the list of tables in the provided namespace.
+
+        Args:
+            namespace (str): the namespace from which to list tables.
+
+        Returns:
+            Sequence[str]: the list of table names.
+        """
+
+        if namespace is not None:
+            return _j_object_list_to_str_list(self.j_object.listTables(namespace))
+        return _j_object_list_to_str_list(self.j_object.listTables())
+
+    def tables_as_table(self, namespace: Optional[str] = None) -> Table:
+        """
+        Returns the list of tables in the provided namespace as a Deephaven table.
+
+        Args:
+            namespace (str): the namespace from which to list tables.
+
+        Returns:
+            a table containing the tables in the provided namespace.
+        """
+
+        if namespace is not None:
+            return Table(self.j_object.listTablesAsTable(namespace))
+        return Table(self.j_object.listTablesAsTable())
+
+    def snapshots(self, table_identifier: str) -> Sequence[str]:
+        """
+        Returns the list of snapshots for the provided table.
+
+        Args:
+            namespace (str): the table from which to list snapshots.
+
+        Returns:
+            the list of snapshots and additional information for each snapshot.
+        """
+
+        snaphot_list = []
+        for snapshot in j_list_to_list(self.j_object.listSnapshots(table_identifier)):
+            snaphot_list.append({
+                "id": snapshot.snapshotId(),
+                "timestamp_ms": snapshot.timestampMillis(),
+                "operation": snapshot.operation(),
+                "summary": snapshot.summary().toString()
+
+            })
+        return snaphot_list
```

Comment:
I'm not sure the correct data structure to return here.  Each `snapshot` has these important properties that the user would like to see. This list is ordered from oldest to newest snapshot.

Any suggestions for this?

Reply:
We haven't wrapped the Java objects, we're just providing the Iceberg library's POJOs. I don't think we should wrap these in Python at all. I think we should just expose our `*AsTable` and `readTable` methods.

Reply:
Should we also rename the functions to remove the `as_tables` distinction? I.e. `namespaces()` returns the namespaces as a table? Same with `tables()` and `snapshots()`?

Reply:
I followed my own suggestion above. It creates a difference in the API for java vs. python so please comment if it worries you.